lib/util/periods.js

```diff
@@ -1180,15 +1180,6 @@ shaka.util.PeriodCombiner = class {
       return false;
     }
 
-    // If the language doesn't match, but the candidate is the "primary"
-    // language, then that should be preferred as a fallback.
-    if (!best.primary && candidate.primary) {
-      return true;
-    }
-    if (best.primary && !candidate.primary) {
-      return false;
-    }
-
```

Comment:
@joeyparrish Removing this completely caused a number of issues, but moving it below the role check solves the issue.